src/d3d12/d3d12_model_pool.hpp

```diff
@@ -78,11 +86,26 @@ namespace wr
 		MemoryBlock* AllocateMemory(MemoryBlock* start_block, std::size_t size, std::size_t alignment);
 		void FreeMemory(MemoryBlock* heap_start_block,  MemoryBlock* block);
 		
-		std::queue<internal::MeshInternal*> m_mesh_stage_queue;
+		std::vector<std::queue<internal::MeshInternal*>> m_mesh_stage_queues;
+
+		std::uint32_t m_current_queue;
 
 		std::uint64_t m_vertex_buffer_size;
 		std::uint64_t m_index_buffer_size;
 
+		util::ThreadPool* m_thread_pool;
```

Comment:
You added a thread pool to the render system and here. You probably only need 1

Reply:
This is the thread pool from the render system. I'm storing the pointer for easier access.